### PR TITLE
Remove mentions of extension inheritance and interfaces

### PR DESCRIPTION
**Pull request by jcouv**

### CODE REVIEW

**Comment by KennethHoff** [comment]
This is worrying; what's the premise? Why's this being removed? 

Is it de-emphasizing implementing interfaces on behalf of another type for the first version, or is this a "we might not ever get this" (.. To a greater extent than the default "we might not ever get this" that all new features inherently bring)?

**Comment by jcouv** [comment]
@KennethHoff I'm scoping down the spec to what we're targeting for first release.

**Comment by KennethHoff** [comment]
Ah, that's what I was assuming, but I was worried, so I had to ask 😅

**Comment by jcouv** [comment]
@AlekseyTs for review. Thanks

File: proposals/extensions.md [hunk 1]
```diff
@@ -166,18 +150,12 @@ extension_modifier
     | 'partial'
     | 'unsafe'
     | 'static'
-    | 'protected'
```
Comment (AlekseyTs) [inline proposals/extensions.md:169]: It looks like the modifiers on a type are modified here. Did you really mean to do that?

**Comment by AlekseyTs** [comment]
Done with review pass (commit 2)

File: proposals/extensions.md (hunk 1 above)
Comment (jcouv) [reply proposals/extensions.md]: That was incorrect. Restored. Thanks

**Review by AlekseyTs — APPROVED** [review]
LGTM (commit 3)

**Pull request merged**